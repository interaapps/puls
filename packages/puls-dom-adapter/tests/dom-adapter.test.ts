import {PulsDOMAdapter} from "../index";
import {createTemplateFunction} from "pulsjs-template";

const html = createTemplateFunction()
test('dom-adapter-test', () => {
    const d = new PulsDOMAdapter(html`test`.parse())

    const [el] = d.render()
    expect(el.textContent).toMatchSnapshot()
})

test('dom-adapter-full', () => {
    const d = new PulsDOMAdapter(html`
        <div>
            <h1>Hello, World!</h1>
            <p>${'test'}</p>
        </div>
    `.parse())

    const [div] = d.render()
    expect(div).toMatchSnapshot()
})
test('dom-adapter-attribute', () => {
    const d = new PulsDOMAdapter(html`
        <div>
            <h1 
                style=${{
                    color: '#FFF'
                }}
                class=${['text-2xl', 'font-bold']}
                id=${'test'}
                normal-value="hello world"
            >
                Hello, World
                !</h1>
            
        </div>
    `.parse())

    const [div] = d.render()
    expect(div).toMatchSnapshot()
})

test('dom-adapter-complex-html', () => {
    const d = new PulsDOMAdapter(html`
        <div>
            <html lang="en">
            <head>
            <meta charset="UTF-8">
                         <meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
                                     <meta http-equiv="X-UA-Compatible" content="ie=edge">
                         <title>Document</title>
            </head>
            <body>
              
            </body>
            </html>
        </div>
    `.parse())

    const [el] = d.render()
    expect(el).toMatchSnapshot()
})


test('dom-adapter-nested', () => {
    const d = new PulsDOMAdapter(html`
        <div>
            ${(new PulsDOMAdapter(html`
                <h1>Hello World!</h1>
            `.parse())).render()}
        </div>
    `.parse())

    const [div] = d.render()
    expect(div).toMatchSnapshot()
})

test('dom-adapter-function-components', () => {

    const TestComponent = (props: {name: string}) => new PulsDOMAdapter(html`
        <div>
            Name is <span id="name">${props.name}</span>
        </div>
    `.parse()).render()

    const d = new PulsDOMAdapter(html`
        <div>
            <${TestComponent} name="JÃ¶n" />
        </div>
    `.parse())

    const [div] = d.render()
    expect(div).toMatchSnapshot()
})

test('dom-adapter-event-handler', () => {
    let clicked = false;
    const d = new PulsDOMAdapter(html`
        <button @click=${() => clicked = true}>Click Me</button>
    `.parse())

    const [button] = d.render()
    button.dispatchEvent(new Event('click'))

    expect(clicked).toBe(true)
    expect(button).toMatchSnapshot()
})

test('dom-adapter-promise', async () => {
    const promiseValue = new Promise(resolve => setTimeout(() => resolve('Async Loaded!'), 100))
    const d = new PulsDOMAdapter(html`
        <div>
            <p>${promiseValue}</p>
        </div>
    `.parse())

    const [div] = d.render()
    expect(div).toMatchSnapshot()

    await promiseValue
    expect(div).toMatchSnapshot()
})
test('dom-adapter-promise-instant', async () => {
    const promiseValue = new Promise(resolve => setTimeout(() => resolve('Async Loaded!')))
    const d = new PulsDOMAdapter(html`
        <div>
            <p>${promiseValue}</p>
        </div>
    `.parse())

    const [div] = d.render()
    expect(div).toMatchSnapshot()

    await promiseValue
    expect(div).toMatchSnapshot()
})

test('dom-adapter-slots', () => {
    const TestComponent = (props: { $slot?: any }) => new PulsDOMAdapter(html`
        <div>
            <h2>Title</h2>
            ${props.$slot}
        </div>
    `.parse()).render()

    const d = new PulsDOMAdapter(html`
        <div>
            <${TestComponent}>
                <p>Inside the component</p>
            </${TestComponent}>
        </div>
    `.parse())

    const [div] = d.render()
    expect(div).toMatchSnapshot()
})
