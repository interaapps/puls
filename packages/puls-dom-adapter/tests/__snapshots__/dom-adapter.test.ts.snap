// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dom-adapter-attribute 1`] = `
<div>
  <h1
    class="text-2xl font-bold"
    id="test"
    normal-value="hello world"
    style="color: rgb(255, 255, 255);"
  >
    Hello, World
                !
  </h1>
</div>
`;

exports[`dom-adapter-complex-html 1`] = `
<div>
  <html
    lang="en"
  >
    <head>
      <meta
        charset="UTF-8"
      />
      <meta
        content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
        name="viewport"
      />
      <meta
        content="ie=edge"
        http-equiv="X-UA-Compatible"
      />
      <title>
        Document
      </title>
    </head>
    <body />
  </html>
</div>
`;

exports[`dom-adapter-full 1`] = `
<div>
  <h1>
    Hello, World!
  </h1>
  <p>
    test
  </p>
</div>
`;

exports[`dom-adapter-function-components 1`] = `
<div>
  <div>
    Name is 
    <span
      id="name"
    >
      JÃ¶n
    </span>
  </div>
</div>
`;

exports[`dom-adapter-nested 1`] = `
<div>
  <h1>
    Hello World!
  </h1>
  
        
</div>
`;

exports[`dom-adapter-test 1`] = `"test"`;
